xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -6,6 +6,23 @@
 
 namespace Slic3r {
 
+int
+PerimeterGenerator::num_loops(int perimeters, int extra_perimeters) {
+    int generated_perimeters = perimeters + extra_perimeters - 1;
+    long int min_shell_thickness, shell_thickness = 0;
```

Comment:
### Style
We don't use `long int`, just `long` (when we aren't using something that typedefs to `long`, like `coord_t`). 

Suggest `auto shell_thickness = this->ext_perimeter_flow.scaled_width()` (yay c++11).